spec/javascripts/live_search_spec.js

```diff
@@ -170,6 +174,44 @@ describe('liveSearch', function () {
     expect(liveSearch.cache('some-slug')).toBe('something in the cache')
   })
 
+  describe('should update the URL correctly', function () {
+    beforeEach(function () {
+      // clear these options to simplify the URLs
+      $form.find('input[name=field]').prop('checked', false)
+      $form.find('input[name=published_at]').val('')
+    })
+
+    it('when search options change', function () {
+      $form.find('input[value="john"]').click()
+      jasmine.Ajax.requests.mostRecent().respondWith({
+        status: 200,
+        response: '{}'
+      })
+
+      expect(window.location.search).toContain('people%5B%5D=john')
+    })
+
+    it('when changed search options result in a cached search', function () {
+      var urls = [
+        'people%5B%5D=john',
+        'people%5B%5D=paul',
+        'people%5B%5D=john&people%5B%5D=paul'
+      ]
+      for (var i = 0; i < urls.length; i++) {
+        jasmine.Ajax.stubRequest('/somewhere.json?' + urls[i]).andReturn({
+          status: 200,
+          response: '{}'
+        })
+      }
+      // set options so that a cached state is triggered
+      $form.find('input[value="john"]').click()
+      $form.find('input[value="paul"]').click()
+      $form.find('input[value="paul"]').click()
+      expect(window.location.search).toContain('people%5B%5D=john')
+      expect(window.location.search).not.toContain('people%5B%5D=paul')
```

Comment:
I'm a bit confused by this test and how it's testing a not cached state. My confusion is based around an expectation I have that this has 3 states (load first results - john, load second results - john & paul, return to cached first result - john) but has 3 stubs, so I'm not sure I'm understanding this wrong.

Reply:
My mistake, I left in a stub that wasn't needed. You're right, this only has two states. Have added some comments to clarify.

Reply:
Thank you!